### DISCUSSION TITLE
Long woff file names for some Quasar web-font libraries

### DISCUSSION THREAD

**Discussion by gtrombitas** [quasarframework/quasar#15842] [General - Components / Directives / etc]
Hi There,

I have a problem that touches some web-font libs in quasar/extras. One of member from our team bundles our Quasar built static web-application on Windows operating system.
The thing is when he packs the folder of the app then he has seen an error in the logs and the files are omitted from the packaging.

Here is the error:

**[path with 132 characters]**/fonts/syl0-zNym6YjUruM-QrEh7-nyTnjDwKNJ_190FjpZIvDmUSVOK7BDB_Qb9vUSzq3wzLK-P0J-V_Zs-QtQth3-jOcDTCVpeRL2w5rwZu2rIekXxE.41201dc5.woff': File or path name too long

After some research I found that this .woff file is in the [extras/material-symbols-rounded](https://github.com/quasarframework/quasar/tree/dev/extras/material-symbols-rounded) package.

My question would be why is this file name soo long? For example [extras/material-icons-round/web-font](https://github.com/quasarframework/quasar/tree/dev/extras/material-icons-round/web-font) and some others have shorter names.

Firstly I don't want to replace the name/path of this file with some hacky webpack methods. Before that it would be great to know the reason of this long name. (E.g. I won't harm any caching behaviour if I change something in my project.)

Thank you in advance.

Regards,
Greg

**Comment by DouglasCalora** [comment]
@gtrombitas Same issue here, any updates?

**Comment by szekelyisz** [comment]
@hawkeye64 I'm facing the same issue. I need to upload my SPA to the flash memory of an embedded device where length of file names is limited. Is there any good reason for such long file names?

**Reply by hawkeye64** [reply]
These names come from Google

**Comment by szekelyisz** [comment]
If using vite, the following `quasar.config.ts` snippet will replace asset file names longer than 48 characters with just the hash of the files' content:

```ts
      extendViteConf: () => ({
        build: {
          rollupOptions: {
            output: {
              assetFileNames: (assetInfo) =>
                assetInfo.names.some((name) => name.length > 48)
                  ? 'assets/[hash][extname]'
                  : 'assets/[name]-[hash][extname]',
            },
          },
        },
      }),
```
